webapp/src/main/java/com/box/l10n/mojito/monitoring/QuartzMetricsReportingJobListener.java

```diff
@@ -0,0 +1,127 @@
+package com.box.l10n.mojito.monitoring;
+
+import com.google.common.base.Strings;
+
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.quartz.JobKey;
+import org.quartz.JobListener;
+import org.quartz.SchedulerException;
+import org.quartz.impl.matchers.GroupMatcher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.data.util.Pair;
+import org.springframework.stereotype.Component;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tags;
+
+import static com.box.l10n.mojito.quartz.QuartzConfig.DYNAMIC_GROUP_NAME;
+
+@ConditionalOnProperty(value = "l10n.management.metrics.quartz-metrics.enabled", havingValue = "true")
+@Component
+public class QuartzMetricsReportingJobListener implements JobListener {
+
+    static Logger logger = LoggerFactory.getLogger(QuartzMetricsReportingJobListener.class);
+
+    private static final String JOB_GROUP = "jobGroup";
+    private static final String JOB_CLASS = "jobClass";
+
+    private MeterRegistry meterRegistry;
+
+    public QuartzMetricsReportingJobListener(@Autowired MeterRegistry meterRegistry) {
+        this.meterRegistry = meterRegistry;
+    }
+
+    @Override
+    public String getName() {
+        return getClass().getSimpleName();
+    }
+
+    @Override
+    public void jobWasExecuted(JobExecutionContext context, JobExecutionException jobException) {
+        try {
+            registerExecutionTime(context);
+            registerPendingJobs(context);
```

Comment:
I agree. I do have mixed feelings as well. As we discussed, the other option here would be to run this method during `jobToBeExecuted`, but at the end of the day, that'll have the exact same effect.

As we discussed, other options would be:

1) Using a `ScheduledTask` to run this query periodically (but we'll need to dedup and probably sanitize metrics being pushed through micrometer)
2) Using a Quartz job to pull these metrics within a scheduled job running in Quartz as well (which can lead to stale metrics or not having metrics at all if quartz is busy processing other jobs)
2.a) On top of this idea, we could play around with job priorities. But with slow & long-running jobs even priorities would create a bottleneck.
2.b) Instead of using priorities, we could create a dedicated scheduler on where we could execute jobs collecting metrics. This would ensure we would collect metrics periodically, but we would still using Quartz to collect metrics.

Another approach would encompass using

Reply:
yes, let's do 1. for now since we don't have that many servers. If we're sending the size queue, we can just do an average.